### ISSUE TITLE
aws_lambda.Code.fromAsset but for EC2

### ISSUE THREAD

**Issue by Mach1212** [aws/aws-cdk#19536]
### Description

I want to deploy 1 EC2 instance with my application being bundled automatically with docker when I deploy. Just like aws_lambda.Code.fromAsset.

### Use Case

I am looking for a single command, cross platform solution, just like with lambda.code.fromAsset. I can sit down on any machine, clone the repo, and run npm install, cdk deploy as long as docker and node are installed. I don't want to setup a pipeline or manually build my application or build a shell script for each os.

### Proposed Solution

It works in lambda, so why not EC2. Am I missing something?

### Acknowledge

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by corymhall** [comment]
@Mach1212 I don't think this would work for deploying an application to an EC2 instance. You would
need to be able to somehow get your code onto an EC2 instance and manage stopping/starting the
application, traffic shifting, etc.

I would recommend looking at a third party library like
[cdk-deployer](https://constructs.dev/packages/cdk-deployer/v/1.0.11?lang=typescript).
